storage/namespace/metadata.go

```diff
@@ -33,16 +35,31 @@ type metadata struct {
 // NewMetadata creates a new namespace metadata
 func NewMetadata(id ts.ID, opts Options) Metadata {
 	copiedID := checked.NewBytes(append([]byte(nil), id.Data().Get()...), nil)
-	return metadata{
+	return &metadata{
```

Comment:
Any reason why this needs to be a pointer instead of a plain struct value? We generally prefer values to pointers to avoid GC sweep overhead.